parsing/dialects/intermediate/dialects/bigquery/parserImpls.ftl

```diff
@@ -15,94 +15,37 @@
 // limitations under the License.
 -->
 
-SqlExcept ExceptExpression(List<SqlNode> selectList):
+SqlExcept ExceptExpression():
 {
     final Pair<SqlNodeList, SqlNodeList> nameAndTypePair;
     final SqlNodeList exceptList;
 }
 {
-    <EXCEPT> nameAndTypePair = ParenthesizedCompoundIdentifierList()
+    <STAR> <EXCEPT> nameAndTypePair = ParenthesizedCompoundIdentifierList()
     {
         exceptList = nameAndTypePair.getKey();
-        if (selectList.size() > 0) {
-            SqlIdentifier identifier = (SqlIdentifier) selectList.get(0);
-            if (!identifier.toString().equals("*") || selectList.size() != 1) {
-                throw SqlUtil.newContextException(getPos(),
-                    RESOURCE.illegalQueryExpression());
-            }
-        }
         return new SqlExcept(getPos(), exceptList);
     }
 }
 
 /**
- * Parses a leaf SELECT expression without ORDER BY.
+ * Parses one unaliased expression in a select list.
  */
-SqlSelect SqlSelect() :
+SqlNode SelectExpression() :
 {
-    final List<SqlLiteral> keywords = new ArrayList<SqlLiteral>();
-    final SqlNodeList keywordList;
-    SqlExcept exceptExpression = null;
-    List<SqlNode> selectList;
-    final SqlNode fromClause;
-    final SqlNode where;
-    final SqlNodeList groupBy;
-    final SqlNode having;
-    final SqlNodeList windowDecls;
-    final List<SqlNode> hints = new ArrayList<SqlNode>();
-    final Span s;
+    SqlNode e;
 }
 {
-    <SELECT>
-    {
-        s = span();
+    LOOKAHEAD(2)
```

Comment:
@michaelvdow Isn't 2 the default LOOKAHEAD value?

Reply:
I think it used to (still might be) but the code didn't work without this.

Reply:
Yeah, it looks like it's still set as 2 in the [build script](https://github.com/googleinterns/calcite/blob/master/parsing/dialects/intermediate/dialects/dialect1/build.gradle.kts#L62)

Reply:
Looks like it is actually set to 1 for bigquery: https://github.com/googleinterns/calcite/blob/master/parsing/dialects/intermediate/dialects/bigquery/build.gradle.kts#L62

Reply:
Aha, good catch. Forgot that it's set to 1 for all others except Dialect1.

Reply:
Ah okay, I thought this was for Dialect1 but that makes sense